lib/dialects/abstract/query-generator.js

```diff
@@ -1520,6 +1522,195 @@ module.exports = (function() {
       return result;
     },
 
+    whereQuery: function(where, options) {
+      var query = this.whereItemsQuery(where, options);
+      if (query && query.length) {
+        return 'WHERE '+query;
+      }
+      return '';
+    },
+    whereItemsQuery: function(where, options, binding) {
+      if (
+        (Array.isArray(where) && where.length === 0) ||
+        (_.isPlainObject(where) && _.isEmpty(where)) ||
+        where === '' ||
+        where === null ||
+        where === undefined
+      ) {
```

Comment:
but aren't you effectively saying every case where "where" is falsy is a no-op case? I agree it seems a little overkill to write it out this way, when all you're trying to convey is "this method will fail if you've provided an empty where: be it object, array or string"